src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -17353,28 +17363,48 @@ def compare(
         new_pandas_labels = []
         new_values = []
         column_index_tuples = []
+        column_types = []
         for (
             pandas_column_value,
             pandas_label,
             left_identifier,
             right_identifier,
             column_only_contains_matches,
+            left_type,
+            right_type,
         ) in zip(
             self.columns,
             filtered_binary_op_result.data_column_pandas_labels,
             self._modin_frame.data_column_snowflake_quoted_identifiers,
             other._modin_frame.data_column_snowflake_quoted_identifiers,
             all_rows_match_frame.iloc[:, 0].values,
+            self._modin_frame.cached_data_column_snowpark_pandas_types,
+            other._modin_frame.cached_data_column_snowpark_pandas_types,
         ):
             # Drop columns that only contain matches.
             if column_only_contains_matches:
                 continue
 
-            cols_equal = col(
-                result_column_mapper.left_quoted_identifiers_map[left_identifier]
-            ).equal_null(
```

Comment:
`equal_null` is used 20 times under `src/snowflake/snowpark/modin/`. I believe we need to update all of them. Am I right?

Reply:
I think we'd have to update most of those places to handle timedelta-number comparisons correctly. We should add test coverage for each case, so I filed SNOW-1641716 to do that separately and I added a TODO pointing to that issue.

Reply:
sounds good.